syft/pkg/cataloger/cpp/package.go

```diff
@@ -8,23 +8,70 @@ import (
 	"github.com/anchore/syft/syft/pkg"
 )
 
-func newConanfilePackage(m pkg.ConanMetadata, locations ...file.Location) *pkg.Package {
-	fields := strings.Split(strings.TrimSpace(m.Ref), "/")
-	if len(fields) < 2 {
-		return nil
+type ConanRef struct {
+	Name      string
+	Version   string
+	User      string
+	Channel   string
+	Revision  string
+	Timestamp string
+}
+
+func splitConanRef(ref string) *ConanRef {
+	// Conan ref format is:
+	// pkg/0.1@user/channel#rrev%timestamp
+	// This method is based on conan's ref.loads method:
+	// https://github.com/conan-io/conan/blob/release/2.0/conans/model/recipe_ref.py#L93C21-L93C21
+
+	var cref ConanRef
+
+	// timestamp
+	tokens := strings.Split(ref, "%")
+	text := tokens[0]
+	if len(tokens) == 2 {
+		cref.Timestamp = tokens[1]
+	} else {
+		cref.Timestamp = ""
 	}
 
-	pkgName, pkgVersion := fields[0], fields[1]
+	// revision
+	tokens = strings.Split(text, "#")
+	ref = tokens[0]
+	if len(tokens) == 2 {
+		cref.Revision = tokens[1]
+	} else {
+		cref.Revision = ""
+	}
 
-	if pkgName == "" || pkgVersion == "" {
+	// name and version are always given
+	tokens = strings.Split(ref, "@")
+	name_version := strings.Split(tokens[0], "/")
+	if name_version[0] == "" || name_version[1] == "" {
 		return nil
 	}
+	cref.Name = name_version[0]
+	cref.Version = name_version[1]
+	// user and channel
+	if len(tokens) == 2 && tokens[1] != "" {
+		tokens = strings.Split(tokens[1], "/")
+		cref.User = tokens[0]
+		cref.Channel = tokens[1]
```

Comment:
This should also check `len(tokens) > 1` before accessing `tokens[1]`

Reply:
This could still be a problem with unexpected input -- before accessing `tokens[1]` the length should be checked, since line 52 is doing another split operation